File: vol127/12725.py
# TLE
for t in range( int( input() ) ):
	n, m, a, b = map( float, input().split() )
	diff = ( m * ( a + b ) - n * a ) / b
	print( 'Case #%d: %s' % ( t + 1, '{:.2f}'.format( diff ) if 0 <= diff <= 10 else 'Impossible' ) )

File: vol127/12725.cpp
#include<cstdio>

int main()
{
	int t;
	double n, m, a, b;

	scanf( "%d", &t );
	for( int x = 1; x <= t && scanf( "%lf %lf %lf %lf", &n, &m, &a, &b ); ++x )
	{
		double diff = ( m * ( a + b ) - n * a ) / b;
		printf( "Case #%d: ", x );
		if( diff >= 0 && diff <= 10 )
			printf( "%.2f\n", diff );
		else
			puts( "Impossible" );
	}
}
